### PR TITLE
Add integ test for Authorization Policy Gateway Policy Attachment

### LINKED ISSUE #47263: Add Integration Test for Authorization Policy with Target Ref
Part of https://github.com/istio/istio/issues/46360
Related to #46847

### PR DESCRIPTION
**Pull request by whitneygriffith**
**Please provide a description of this PR:**

Resolves #47263

### CODE REVIEW

File: tests/integration/security/policy_attachment_only/jwt_gateway_test.go [hunk 1]
```diff
@@ -190,6 +190,86 @@ func TestGatewayAPIRequestAuthentication(t *testing.T) {
 		})
 }
 
+func TestGatewayAPIAuthorizationPolicy(t *testing.T) {
+	framework.NewTest(t).
+		Label(label.IPv4). // https://github.com/istio/istio/issues/35835
+		Features("security.authorization.gateway-api").
+		Run(func(t framework.TestContext) {
+			crd.DeployGatewayAPIOrSkip(t)
+			config.New(t).
```
Comment (MorrisLaw) [inline tests/integration/security/policy_attachment_only/jwt_gateway_test.go:199]: Are the configs being built and applied correctly? Are these params correct?

Comment (whitneygriffith) [reply tests/integration/security/policy_attachment_only/jwt_gateway_test.go:199]: Yes the building is set up correctly. I believe the error lies with the rules in my authorization policy config. I have modified it.

File: tests/integration/security/policy_attachment_only/jwt_gateway_test.go [hunk 2]
```diff
@@ -190,6 +190,98 @@ func TestGatewayAPIRequestAuthentication(t *testing.T) {
...
+							opts.HTTP.Method = "GET"
+							opts.HTTP.Headers = headers.New().
+								WithHost(fmt.Sprintf("example.%s.com", to.ServiceName())).
+								WithAuthz(jwt.TokenIssuer1).
+								Build()
+							opts.Check = check.OK()
+						},
+					},
+					{
+						name: "deny without token",
```
Comment (jaellio) [inline tests/integration/security/policy_attachment_only/jwt_gateway_test.go:229]: Do we need a test case for denying with an incorrect token?

File: tests/integration/security/policy_attachment_only/testdata/authz/gateway-authz.yaml.tmpl [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    - operation:
+        hosts:
+{{- range $svc := .Services }}
+        - "example.{{ $svc.ServiceName }}.com"
+{{- end }}
+    from:
+    - source:
+        requestPrincipals: ["test-issuer-1@istio.io/sub-1"]
+    - source:
+        requestPrincipals: ["test-issuer-3@istio.io/sub-1"]
```
Comment (jaellio) [inline tests/integration/security/policy_attachment_only/testdata/authz/gateway-authz.yaml.tmpl:37]: What is the significance of test-issuer-3?

File: tests/integration/security/policy_attachment_only/testdata/authz/gateway-api.yaml.tmpl [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  - "*.{{ .To.ServiceName }}.com"
+  rules:
+  - matches:
+    - path: 
+        type: PathPrefix
+        value: /allow
+      method: GET
+    backendRefs:
+    - name: "{{ .To.ServiceName }}"
+      port: {{ (.To.PortForName `http`).ServicePort }}
```
Comment (jaellio) [inline tests/integration/security/policy_attachment_only/testdata/authz/gateway-api.yaml.tmpl:30]: nit: missing newline

File: tests/integration/security/policy_attachment_only/testdata/authz/gateway-authz.yaml.tmpl (hunk 3 above)
Comment (whitneygriffith) [reply tests/integration/security/policy_attachment_only/testdata/authz/gateway-authz.yaml.tmpl:37]: it isn't used so I will remove it

File: tests/integration/security/policy_attachment_only/jwt_gateway_test.go (hunk 2 above)
Comment (whitneygriffith) [reply tests/integration/security/policy_attachment_only/jwt_gateway_test.go:229]: I think its good to have as it confirms the `AuthPolicy` is enforcing only requests with valid token. Happy to remove if others think its not necessary.

**Comment by whitneygriffith** [comment]
> Looks good. Do we plan to support listener attachment too ?

do you mind expanding on what listener attachment is? 

> One small issue - if someone looks at the tests as an example, they may get the wrong impression. I would add a small comment to the gateway yaml - jwts are not best practice on port 80 ( some RFCs don't allow them).

updated!

**Comment by MorrisLaw** [comment]
> Looks good. Do we plan to support listener attachment too ?
> 
> One small issue - if someone looks at the tests as an example, they may get the wrong impression. I would add a small comment to the gateway yaml - jwts are not best practice on port 80 ( some RFCs don't allow them).

Hey @costinm, wdym by listener attachment? Is it [this ](https://gateway-api.sigs.k8s.io/geps/gep-713/#apply-policies-to-sections-of-a-resource)or something else?

**Review by MorrisLaw — APPROVED** [review]
👍

**Review by howardjohn — APPROVED** [review]

File: tests/integration/security/policy_attachment_only/jwt_gateway_test.go (hunk 2 above)
Comment (jaellio) [reply tests/integration/security/policy_attachment_only/jwt_gateway_test.go:229]: I was curious if we wanted a test case with an incorrect token in addition to the test case with no token.

Comment (whitneygriffith) [reply tests/integration/security/policy_attachment_only/jwt_gateway_test.go:229]: Oh I misunderstood!

**Pull request merged**